src/main/java/competition/subsystems/vision/VisionSubsystem.java

```diff
@@ -131,7 +134,9 @@ public Optional<EstimatedRobotPose> getPhotonVisionEstimatedPose(Pose2d previous
             //return customPhotonPoseEstimator.update();
             photonPoseEstimator.setReferencePose(previousEstimatedRobotPose);
             var estimatedPose = photonPoseEstimator.update();
-            if (!estimatedPose.isEmpty() && this.isEstimatedPoseReliable(estimatedPose.get())) {
+            var isReliable = isEstimatedPoseReliable(estimatedPose.get());
+            var isStable = frontReliablePoseIsStable.checkStable(isReliable);
+            if (!estimatedPose.isEmpty() && isReliable && isStable) {
```

Comment:
I think it's unecessary to check `!estimatedPose.isEmpty()` here again? It can't be reliable  if it's empty?

Reply:
The Pose tests failed because estimatedPose.get() throws if it's empty. I had to move this.